From 784d44ca70fee6a576064c4fcf6f54562f5b7117 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 2 Oct 2025 17:29:10 +0200
Subject: [PATCH] arm64: dts: stm32: add sleep pin configuration for goodix on
 stm32mp215f-dk

Temporary patch to add sleep pin configuration for goodix touchscreen
on stm32mp215f-dk.

Change-Id: Iad5ce8df4baa84de458ed0430f6ec07ee9fbcd8f
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/495798
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi | 7 +++++++
 arch/arm64/boot/dts/st/stm32mp215f-dk.dts     | 3 ++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi b/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi
index c83b31aec240..3c4e6ed01f0e 100644
--- a/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi
@@ -64,6 +64,13 @@ pins2 {
 		};
 	};
 
+	goodix_sleep_pins_a: goodix_sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('I',  8, ANALOG)>,
+				 <STM32_PINMUX('F',  4, ANALOG)>;
+		};
+	};
+
 	i2c2_pins_a: i2c2-0 {
 		pins {
 			pinmux = <STM32_PINMUX('B', 5, AF9)>, /* I2C2_SCL */
diff --git a/arch/arm64/boot/dts/st/stm32mp215f-dk.dts b/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
index 1ea9904a7064..94218c32c043 100644
--- a/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
@@ -382,8 +382,9 @@ &i2c2 {
 	goodix: goodix-ts@5d {
 		compatible = "goodix,gt911";
 		reg = <0x5d>;
-		pinctrl-names = "default";
+		pinctrl-names = "default", "sleep";
 		pinctrl-0 = <&goodix_pins_a>;
+		pinctrl-1 = <&goodix_sleep_pins_a>;
 		interrupt-parent = <&gpiof>;
 		interrupts = <4 IRQ_TYPE_EDGE_FALLING>;
 		AVDD28-supply = <&scmi_v3v3>;
-- 
2.53.0

